### PR TITLE
add swr for decoupling requests, add search results page

### PR DESCRIPTION
**Pull request by mentealterne**
Hello Luigi,
sorry for the long PR, I've introduced SWR for decoupling requests, done some refactoring to the UI code and also added a dedicated search page so that search  results links can be shareable. Have an happy sunday, it's mid-day so we can start drinking some wine !

### CODE REVIEW

**Comment by luigibarbato** [comment]
SBAM! @mentealterne 💣 

Thank you so much for this amazing enhancement, Love it! ❤️ 

However, I would like to limit the graphic modifications to the components for now. In my opinion, we should focus more on the main structure and functionality since we don't have any strong ideas about restyling the components at the moment. I plan to dedicate an entire chapter to component restyling in the near future. Once again, Alfre, thank you very much 🥇 .

File: pages/index.tsx [hunk 1]
```diff
@@ -1,136 +1,9 @@
...
-        )}
-        <main className="m-auto md:mb-0 md:pb-10 overflow-y-scroll max-h-screen md:overflow-y-visible md:max-h-full">
-          {loading && <Loading />}
-          <Feeds feeds={feeds} />
-        </main>
-        <Footer />
-      </div>
-    </>
+    <div className={"w-full h-full flex flex-col items-center justify-center"}>
+      <h1>You are the only one who can decide what to think.</h1>
```
Comment (luigibarbato) [inline pages/index.tsx:5]: Leftover? :D

Comment (mentealterne) [reply pages/index.tsx:5]: Hello Louis, not properly a leftover, just a placeholder to not leave home page empty waiting for ui restyling !

**Review by luigibarbato — APPROVED** [review]
SBAM 🔥 

Thank you again, @mentealterne 💣

**Pull request merged**